tests/Unit/Cache/Psr16Test.php

```diff
@@ -0,0 +1,240 @@
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @package SimplePie
+ * @copyright 2004-2022 Ryan Parman, Sam Sneddon, Ryan McCue
+ * @author Ryan Parman
+ * @author Sam Sneddon
+ * @author Ryan McCue
+ * @link http://simplepie.org/ SimplePie
+ * @license http://www.opensource.org/licenses/bsd-license.php BSD License
+ */
+
+namespace SimplePie\Tests\Unit\Cache;
+
+use Exception;
+use PHPUnit\Framework\TestCase;
+use Psr\SimpleCache\CacheInterface;
+use SimplePie\Cache\Psr16;
+use SimplePie\Tests\Fixtures\Exception\Psr16CacheException;
+
+class Psr16Test extends TestCase
+{
+    public function testImplementationIsGloballyStored()
+    {
+        $this->expectException(Exception::class);
+        $this->expectExceptionMessage('You must set an implementation of `Psr\SimpleCache\CacheInterface` via `SimplePie\SimplePie::set_cache()` first.');
+
+        new Psr16('location', 'name', 'type');
+    }
+
+    public function testSaveReturnsTrue()
+    {
+        $data = [];
+        $psr16 = $this->createMock(CacheInterface::class);
+        $psr16->expects($this->exactly(2))->method('set')->withConsecutive(
+            [
+                '18bb0a0a94b49eda35e8944672d60a1474ff2895524f0e56c3752c1e0f7853e7',
+                $data,
+                3600,
+            ],
+            [
+                '18bb0a0a94b49eda35e8944672d60a1474ff2895524f0e56c3752c1e0f7853e7_mtime'
+            ]
+        )->willReturn(true);
+
+        Psr16::store_cache($psr16);
+        $cache = new Psr16('location', 'name', 'type');
+
+        $this->assertTrue($cache->save($data));
+    }
+
+    public function testSaveReturnsFalse()
+    {
+        $data = [];
+        $psr16 = $this->createMock(CacheInterface::class);
+        $psr16->expects($this->exactly(2))->method('set')->willReturn(false);
+
+        Psr16::store_cache($psr16);
+        $cache = new Psr16('location', 'name', 'type');
+
+        $this->assertFalse($cache->save($data));
+    }
+
+    public function testSaveCatchExceptionAndReturnsFalse()
+    {
+        $e = new Psr16CacheException();
+
+        $data = [];
+        $psr16 = $this->createMock(CacheInterface::class);
+        $psr16->expects($this->once())->method('set')->willThrowException($e);
+
+        Psr16::store_cache($psr16);
+        $cache = new Psr16('location', 'name', 'type');
+
+        $this->assertFalse($cache->save($data));
+    }
+
+    public function testLoadReturnsData()
+    {
+        $data = [];
+        $psr16 = $this->createMock(CacheInterface::class);
+        $psr16->expects($this->once())->method('get')->willReturn($data);
+
+        Psr16::store_cache($psr16);
+        $cache = new Psr16('location', 'name', 'type');
+
+        $this->assertSame($data, $cache->load());
+    }
+
+    public function testLoadReturnsFalse()
+    {
+        $psr16 = $this->createMock(CacheInterface::class);
+        $psr16->expects($this->once())->method('get')->willReturnCallback(function ($a, $b) {
```

Comment:
Maybe add a comment in the vein of “Returning the sentinel passed as the default value to simulate a cache miss in the backend.” And use `$key` and `$default` as argument names to match PSR-16.

Reply:
I updated the name of the test case to make clear that a cache miss is tested.